### PR TITLE
feat(components/atom/button): Add elevation prop and focus var

### PR DESCRIPTION
**Pull request by andresz1**
### Types of changes
- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Description, Motivation and Context
This PR:
- Add `elevation` prop that can be used to display a shadow around the button. If it's not set no shadow will be shown
- Add `focus` custom shadow. By default is set to `none` so no breaking change is performed. Nevertheless each vertical should override `$bxsh-atom-butto-focus` variable since focus is really important in terms of accessibility (e.g. to use keyword navigation the user should visually know which element is focused)

### Screenshots - Animations

Documentation
Screenshot 2022-08-17 at 16 33 15

Focus
Screenshot 2022-08-17 at 16 03 37

### CODE REVIEW

File: components/atom/button/README.md [hunk 1]
```diff
@@ -142,21 +169,33 @@ return (
...
-
 ```
 
 output:
 
 ```html
-<a class="sui-AtomButton sui-AtomButton--link" href="http://www.schibsted.com/" target='_blank'>Link</a>
+<a
+  class="sui-AtomButton sui-AtomButton--link"
+  href="http://www.schibsted.com/"
```
Comment (turolopezsanabria) [inline components/atom/button/README.md]:
```suggestion
  href="https://www.adevinta.com/"
```

File: components/atom/button/src/styles/index.scss [hunk 2]
```diff
@@ -244,6 +244,24 @@ $base-class: '.sui-AtomButton';
...
+
+        &#{$base-class}--negative {
+          box-shadow: $bxsh;
+        }
+      }
+    }
+  }
+
+  &:focus {
+    box-shadow: $bxsh-atom-butto-focus;
```
Comment (turolopezsanabria) [inline components/atom/button/src/styles/index.scss]:
```suggestion
    box-shadow: $bxsh-atom-button-focus;
```

File: components/atom/button/src/styles/settings.scss [hunk 3]
```diff
@@ -99,6 +99,12 @@ $p-atom-button-group-sides: $p-l !default;
 $bxsh-atom-button-solid-primary: none !default;
 $bxsh-atom-button-solid-primary-negative: none !default;
 
+$bxsh-atom-button: (
+  large: $bxsh-l,
+  medium: $bxsh-m
+) !default;
+$bxsh-atom-butto-focus: none !default;
```
Comment (turolopezsanabria) [inline components/atom/button/src/styles/settings.scss]:
```suggestion
$bxsh-atom-button-focus: none !default;
```

**Comment by andresz1** [comment]
@turolopezsanabria changes made! thanks for the feedback

**Review by turolopezsanabria — APPROVED** [review]
thnx!

**Review by javiauso — APPROVED** [review]

**Review by oriolpuig — APPROVED** [review]

**Pull request merged**